nitro-protocol/contracts/MultiAssetHolder.sol

```diff
@@ -41,37 +41,23 @@ contract MultiAssetHolder is IMultiAssetHolder, StatusManager {
         uint256 amount
     ) external payable virtual override {
         require(!_isExternalDestination(channelId), 'Deposit to external destination');
-        uint256 amountDeposited;
         // this allows participants to reduce the wait between deposits, while protecting them from losing funds by depositing too early. Specifically it protects against the scenario:
         // 1. Participant A deposits
         // 2. Participant B sees A's deposit, which means it is now safe for them to deposit
         // 3. Participant B submits their deposit
         // 4. The chain re-orgs, leaving B's deposit in the chain but not A's
         uint256 held = holdings[asset][channelId];
-        require(held >= expectedHeld, 'holdings < expectedHeld');
-        require(held < expectedHeld + amount, 'holdings already sufficient');
+        require(held == expectedHeld, 'held != expectedHeld');
 
-        // The depositor wishes to increase the holdings against channelId to amount + expectedHeld
-        // The depositor need only deposit (at most) amount + (expectedHeld - holdings) (the term in parentheses is non-positive)
-
-        amountDeposited = expectedHeld + amount - held; // strictly positive
         // require successful deposit before updating holdings (protect against reentrancy)
         if (asset == address(0)) {
             require(msg.value == amount, 'Incorrect msg.value for deposit');
         } else {
-            IERC20(asset).safeTransferFrom(msg.sender, address(this), amountDeposited);
+            IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
         }
 
-        uint256 nowHeld = held + amountDeposited;
-        holdings[asset][channelId] = nowHeld;
-        emit Deposited(channelId, asset, amountDeposited, nowHeld);
-
-        if (asset == address(0)) {
-            // refund whatever wasn't deposited.
-            uint256 refund = amount - amountDeposited;
-            (bool success, ) = msg.sender.call{value: refund}(''); //solhint-disable-line avoid-low-level-calls
-            require(success, 'Could not refund excess funds');
-        }
+        holdings[asset][channelId] += amount;
+        emit Deposited(channelId, asset, amount, holdings[asset][channelId]);
```

Comment:
Nit: I think we may be able to save some gas by reducing the number of storage reads? Currently there are 3, and I think in principle it should be do-able in 1. Might be worth an investigation (no need to hold up this PR though).

Reply:
:+1: Tracked in #1410